### PR TITLE
Fixed the failure from strict mode.

### PR DESCRIPTION
**Pull request by sima-zhu**
We got several issues when enable the strict mode from scripts.
JS - mgmt pipeline: Failed at $LastExitCode not retrieved.
[Pipeline](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1343628&view=logs&j=f8e040b3-c0ff-5789-0eda-99daaaa3fc1b&t=26ac1571-47fc-5d52-3897-39c8bdecf4db)

Doc msauthor failed by `Write-Error`, change to `Write-Warning`
[Pipeline](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1343627&view=logs&j=f8e040b3-c0ff-5789-0eda-99daaaa3fc1b&t=4ddc76a6-85f2-526e-fa3e-b1020f31ed12)

### CODE REVIEW

File: eng/common/scripts/Update-DocsMsMetadata.ps1 [hunk 1]
```diff
@@ -225,10 +225,6 @@ foreach ($packageInfoLocation in $PackageInfoJsonLocations) {
   if ($ValidateDocsMsPackagesFn -and (Test-Path "Function:$ValidateDocsMsPackagesFn")) {
     Write-Host "Validating the package..."
     &$ValidateDocsMsPackagesFn -PackageInfo $packageInfo -PackageSourceOverride $PackageSourceOverride -DocValidationImageId $DocValidationImageId -DocRepoLocation $DocRepoLocation
-    if ($LASTEXITCODE) {
```
Comment (weshaggard) [inline eng/common/scripts/Update-DocsMsMetadata.ps1:228]: Please be sure to file a follow-up issue which will figure out the correct way to handle validation errors.

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/Update-DocsMsMetadata.ps1 (hunk 1 above)
Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsMetadata.ps1:228]: Found someone report an issue while I draft an issue to fix this.
https://github.com/Azure/azure-sdk-tools/issues/2669

**Review by danieljurek — APPROVED** [review]
This is a good short term mitigation. If a package fails validation and isn't caught here the risk is that the readme page will update to the new version but the package may not necessarily be onboarded at that new version when docindex runs. 


Indeed `$LASTEXITCODE` is not set because these run in a `ForEach-Object -Parallel` https://github.com/Azure/azure-sdk-for-js/blob/main/eng/scripts/Language-Settings.ps1#L243 

Is there a bug tracking converting the JS validation to only validate updated packages instead of running validation on all packages at the end? https://github.com/Azure/azure-sdk-for-js/blob/main/eng/scripts/Language-Settings.ps1#L377

File: eng/common/scripts/Update-DocsMsMetadata.ps1 (hunk 1 above)
Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsMetadata.ps1:228]: File an issue tracking the followup work
https://github.com/Azure/azure-sdk-tools/issues/2676

**Pull request merged**